plugins/devices/restapi-v3/src/main/java/com/freedomotic/plugins/devices/restapiv3/resources/jersey/UserResource.java

```diff
@@ -86,16 +88,30 @@ public Response create(UserRepresentation s) {
     @Override
     protected URI doCreate(UserRepresentation o) throws URISyntaxException {
         User u = new User(o.getName(), o.getPassword(), API.getAuth());
-        u.setRoles(o.getRoles());
-        for (Object key : o.getProperties().keySet()) {
-            u.setProperty(key.toString(), o.getProperties().getProperty(key.toString()));
+
+        Set<String> roles = o.getRoles();
+        if (roles == null || roles.isEmpty()) {
```

Comment:
Why don't use `Optional `here?

Reply:
I considered using `Optional` here and it would work well. Only thing stopping me was that I did not want to modify existing model too much, especially if it was part of the API model. But if you prefer such approach I'd be glad to do it `Optional` way.

Reply:
@mcicolella opinions?

Reply:
Maybe the best solution is having a single endpoint removing the schema property and having username and password as mandatory. Others could be optional. By default role could be "guest" with minimal privileges and only an administrator should change it for security reasons.